### PR TITLE
bug(BILL-CPC-826): Integrated BillRequest DTO and BillResponseDTO into code

### PR DESCRIPTION
**Pull request by drallo22**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/35?modal=detail&selectedIssue=CPC-826&sprint=290

## Context:
Worked on updating the DTO's as well as making changes and creating tests. Slight changes to UI
## Changes
- Changed all code that uses BillDTO to use BillRequestDTO and BillResponseDTO (except the update method as Minseok will implement that for his story)
- Split BillDetails (in API-GATEWAY) to BillRequestDTO and BillResponseDTO and implemented this throughout the code
- Changed all tests that use BillDTO to use BillRequestDTO and BillResponseDTO 
- Created BillDomainClientIntegrationTest in API-Gateway and implemented tests
- Updated Bill detail page to include link to get all bills by a vet Id
- Updated the frontend to implement reactive flux producing text_event_stream_value with the frontend for get bill history, bills by customer and bills by vet

## Before and After UI (Required for UI-impacting PRs)
**Before**
image

**After**
Screenshot 2023-09-19 142339
Screenshot 2023-09-19 143002

**Side Note**
UI may have missing names, which is being fixed in other stories.

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java [hunk 1]
```diff
@@ -57,31 +59,31 @@ public class BFFApiGatewayController {
...
 
     @PostMapping(value = "bills",
             consumes = "application/json",
             produces = "application/json")
-    public Mono<BillDetails> createBill(@RequestBody BillDetails model) {
+    public Mono<BillResponseDTO> createBill(@RequestBody BillRequestDTO model) {
         return billServiceClient.createBill(model);
     }
 
     @GetMapping(value = "bills")
```
Comment (swafit) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:80]: Fluxes should include produces TEXT_EVENT_STREAM_VALUE

File: api-gateway/src/main/resources/static/scripts/bill-history/bill-history.controller.js
Comment (swafit) [inline api-gateway/src/main/resources/static/scripts/bill-history/bill-history.controller.js:1]: Good implementation of the EventSource

**Review by ebond07 — APPROVED** [review]
That's a lot of changing BillDetails and BillDTO to BillRequestDTO and BillResponseDTO. Also test looks good.

File: api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java (hunk 1 above)
Comment (dazhuox) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java:80]: These are Monos

**Review by dazhuox — APPROVED** [review]
Good job on the correct/up-to-date way to implement DTOs.

**Pull request merged**